File: backend/src/main/java/com/electoral/entities/ReglaNegocio.java
package com.electoral.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import java.time.LocalDateTime;

@Entity
@Table(name = "reglas_negocio")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReglaNegocio {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 20)
    private String tipo;

    @Column(nullable = false, length = 30)
    private String modulo;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "condicion", nullable = false)
    private String condicion;

    @Column(name = "mensaje_error", length = 500)
    private String mensajeError;

    @Column(length = 50)
    private String accion;

    @Builder.Default
    @Column(nullable = false)
    private Boolean activa = true;

    @Builder.Default
    @Column(nullable = false)
    private Integer prioridad = 0;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: conteo_electoral_app/lib/models/regla_negocio.dart
class ReglaNegocio {
  final int id;
  final String tipo;
  final String modulo;
  final String nombre;
  final String? descripcion;
  final String condicion;
  final String? mensajeError;
  final String? accion;
  final bool activa;
  final int prioridad;
  final String? createdAt;
  final String? updatedAt;

  ReglaNegocio({
    required this.id,
    required this.tipo,
    required this.modulo,
    required this.nombre,
    this.descripcion,
    required this.condicion,
    this.mensajeError,
    this.accion,
    required this.activa,
    required this.prioridad,
    this.createdAt,
    this.updatedAt,
  });

  factory ReglaNegocio.fromJson(Map<String, dynamic> json) {
    return ReglaNegocio(
      id: json['id'],
      tipo: json['tipo'] ?? 'VALIDACION',
      modulo: json['modulo'] ?? 'CANDIDATOS',
      nombre: json['nombre'] ?? '',
      descripcion: json['descripcion'],
      condicion: json['condicion'] ?? '{}',
      mensajeError: json['mensajeError'],
      accion: json['accion'],
      activa: json['activa'] ?? true,
      prioridad: json['prioridad'] ?? 0,
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tipo': tipo,
      'modulo': modulo,
      'nombre': nombre,
      'descripcion': descripcion,
      'condicion': condicion,
      'mensajeError': mensajeError,
      'accion': accion,
      'activa': activa,
      'prioridad': prioridad,
    };
  }
}
